### PR TITLE
New issue: Precondition on is_convertible is too strong

### PR DESCRIPTION
**Pull request by CaseyCarter**

### CODE REVIEW

**Comment by mclow** [comment]
Hey Casey, what's a stronk? ;-)

**Comment by CaseyCarter** [comment]
> When is `add_rvalue_reference_t<Foo>` neither a complete type nor *cv* `void`?

Yeah, I know - it felt weird to say nothing about `From` in the precondition. But requiring something that is necessarily true is bizarre - I'll change this.

**Comment by CaseyCarter** [comment]
There's a fresh push with a new issue number and no "stronk".

File: xml/issue3174.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+relax the precondition to allow this perfectly valid case (and similar cases
+like <tt>is_convertible&lt;S, S&amp;&amp;&gt;</tt>), letting the cases that
+would in fact be invalid fall through to the blanket "incompletely-defined
+object type" wording in
+<a href="https://wg21.link/meta.rqmts#5">[meta.rqmts]/5</a>.
+</p>
+</discussion>
+
+<resolution>
+<p>This wording is relative to <a href="https://wg21.link/n4788">N4788</a>.</p>
```
Comment (timsong-cpp) [inline xml/issue3174.xml]: N4791.

Comment (Dani-Hub) [reply xml/issue3174.xml]: I checked the wording state against N4791 and bumped the reference working draft in the issue now, see http://cplusplus.github.io/LWG/lwg-active.html#3174

**Pull request merged**